File: src/main/scala/io/lamma/DayOfWeek.scala
package io.lamma

sealed trait DayOfWeek {
  def ordinal = DayOfWeek.DayOfWeeks.indexOf(this) + 1
}

object DayOfWeek {
  final val DayOfWeeks = List(Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday)

  /**
   *
   * @param n Monday == 1
   * @return
   */
  def apply(n: Int) = DayOfWeeks(n - 1)
}

case object Monday extends DayOfWeek

case object Tuesday extends DayOfWeek

case object Wednesday extends DayOfWeek

case object Thursday extends DayOfWeek

case object Friday extends DayOfWeek

case object Saturday extends DayOfWeek

case object Sunday extends DayOfWeek

File: src/main/java/io/lamma/DayOfWeek.java
package io.lamma;

public enum DayOfWeek {

    MONDAY,

    TUESDAY,

    WEDNESDAY,

    THURSDAY,

    FRIDAY,

    SATURDAY,

    SUNDAY;

    /**
     * nth day of a week, start from MONDAY. <br/>
     *  MONDAY => 1
     *  TUESDAY => 2
     *  WEDNESDAY => 2
     *  THURSDAY => 2
     *  FRIDAY => 2
     *  SATURDAY => 2
     *  SUNDAY => 2
     */
    public int n() {
        return this.ordinal() + 1;
    }

    public static DayOfWeek of(int i) {
        return values()[i - 1];
    }
}
